report/2.Analysis/HighOrderFunctions.tex

```diff
@@ -0,0 +1,56 @@
+\section{High-order functions}
+
+High-order functions is a concept that were discussed and valued as an important key feature of HCL.
+High-order functions allows for some powerful stuff to be added to the language.
+Along with this, high-order functions also may drastically improve development time, since it allows for more generic functions that can be concretized through the high-order functions.
+
+\subsection{Explanation of high-order functions}
+Basically high-order functions are functions that either returns a function or takes a functions as one of its parameters. 
+This language feature is often utilized together with the three functions map, filter and reduce. 
+To understand what these does, and why they are useful have a look at the example pseudo code in \ref{lis:hofExamples}. 
+One thing to note, is that the function parameters in the example is passed as lambda functions. 
+Lambda functions are anonymous inline functions, which are very useful along with high-order functions, since it will allow you to specify the functionality without having to declare an entire function.
+
+\begin{lstlisting}[language=Python,label=lis:hofExamples,caption=An example of the map\, filter and reduce functions.,firstnumber=1]
+myList = [5, 10, 15, 20, 25]
+
+myNewNum = myList.filter{ num => num % 10 == 0 }.map{ num => num * x }.reduce{ sumOfNums, num => sumOfNums + num }
+
+print(myNewNum) // 500
+\end{lstlisting}
+
+First a list with the numbers 5, 10, 15, 20 and 25 is initialized. 
+Then the 3 functions are chained to get the sum of all these numbers squared, where the initial number was a divisor of 10.
```

Comment:
Change to: The 3 functions are then chained

Reply:
Linje 9: Skriv functionsnavne med \texttt{navn}
Linje 10: "does" til "do". Skriv hvad man refererer til, så PDF'en ikke siger "pseudo code in 2.4"
Linje 11: Fjern komma efter "note". "example is" skal være "example are"
Linje 12: "since it" skal være "since they"